### ISSUE TITLE
k8s-gsg-test is failing due to Cilium daemon crashing

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#1084]
```vagrant@vagrant:~$ kubectl logs cilium-5csw7   -n kube-system
time="2017-07-05T19:15:34Z" level=info msg="     _ _ _" 
time="2017-07-05T19:15:34Z" level=info msg=" ___|_| |_|_ _ _____" 
time="2017-07-05T19:15:34Z" level=info msg="|  _| | | | | |     |" 
time="2017-07-05T19:15:34Z" level=info msg="|___|_|_|_|___|_|_|_|" 
time="2017-07-05T19:15:34Z" level=info msg="Cilium 0.9.90 3f13b0d Tue, 4 Jul 2017 13:54:39 +0200 go version go1.8.3 linux/amd64" 
time="2017-07-05T19:15:34Z" level=info msg="clang and kernel versions: OK!" 
time="2017-07-05T19:15:34Z" level=info msg="linking environment: OK!" 
time="2017-07-05T19:15:36Z" level=info msg="BPF Verifier: NOT OK:\nHAVE_SKB_CHANGE_TAIL: Kernel does not support bpf_skb_change_tail() helper. Therefore, cilium does not support sending ICMPv6 time exceeded notifications. Recommendation is to run 4.9+ kernels.\nHAVE_LPM_MAP_TYPE: Your kernel doesn't support LPM trie maps for BPF, thus disabling CIDR policies. Recommendation is to run 4.11+ kernels.\nHAVE_LRU_MAP_TYPE: Your kernel doesn't support LRU maps for BPF, thus switching back to using hash table for the cilium connection tracker. Recommendation is to run 4.10+ kernels.\nHAVE_MAP_VAL_ADJ: Verifier is too old to detect dynamic map value access after bpf_map_lookup_elem(). Recommendation is to run 4.9+ kernels.\nHAVE_MARK_MAP_VALS: Verifier is too old to detect identical registers with map value after bpf_map_lookup_elem(). Some clang versions might generate code that spills such registers to stack before a NULL test. Recommendation is to run 4.9.17+ kernels.\n" 
time="2017-07-05T19:15:36Z" level=info msg="Consul client ready" 
time="2017-07-05T19:15:36Z" level=info msg="Using consul as key-value store" 
time="2017-07-05T19:15:36Z" level=info msg="Valid label prefix configuration:" 
time="2017-07-05T19:15:36Z" level=info msg=" - :io.kubernetes.pod.namespace" 
time="2017-07-05T19:15:36Z" level=info msg=" - !:io.kubernetes" 
time="2017-07-05T19:15:36Z" level=info msg=" - !:.*kubernetes.io" 
time="2017-07-05T19:15:36Z" level=info msg=" - !:pod-template-hash" 
time="2017-07-05T19:15:36Z" level=debug msg="Creating 1 worker threads" 
time="2017-07-05T19:15:36Z" level=fatal msg="Error while creating daemon: unable to retrieve node CIDR: Get http://127.0.0.1:8080/api/v1/nodes/vagrant: dial tcp 127.0.0.1:8080: getsockopt: connection refused" 
```

**Comment by ianvernon** [comment]
Looks like the test is using K8s version 1.7 now:
```
vagrant@vagrant:~$ kubectl get nodes              
NAME      STATUS    AGE       VERSION
vagrant   Ready     23m       v1.7.0
```
will try with 1.6 and see if this continues to occur

**Comment by ianvernon** [comment]
Tests have passed on master recently using 1.7 versions of K8s: https://jenkins.cilium.io/job/cilium/job/cilium/job/master/397/execution/node/23/log/?consoleFull
: 
```
==> default: Get:5 http://us.archive.ubuntu.com/ubuntu yakkety/universe amd64 socat amd64 1.7.3.1-1 [321 kB]
==> default: Get:1 https://packages.cloud.google.com/apt kubernetes-xenial/main amd64 kubernetes-cni amd64 0.5.1-00 [5,560 kB]
==> default: Get:2 https://packages.cloud.google.com/apt kubernetes-xenial/main amd64 kubelet amd64 1.7.0-00 [19.4 MB]
==> default: Get:3 https://packages.cloud.google.com/apt kubernetes-xenial/main amd64 kubectl amd64 1.7.0-00 [10.1 MB]
==> default: Get:4 https://packages.cloud.google.com/apt kubernetes-xenial/main amd64 kubeadm amd64 1.7.0-00 [9,786 kB]
```

**Comment by ianvernon** [comment]
Looks like it started failing in master at this build: https://jenkins.cilium.io/job/cilium/job/cilium/job/master/398/console .

Commit associated with the build: 
```
commit 549bd4be5ecf76c2cb6d23e0a4824593dc4a4a6f
Author: Thomas Graf <thomas@cilium.io>
Date:   Tue Jul 4 03:19:47 2017 +0200

    k8s-tests: Use files of local git tree
    
    - Do not run testsuite against DaemonSet, policy and demo yaml of git
      master. Use the local checked out git tree instead.
    
    - Be less verbose when running test
    
    Signed-off-by: Thomas Graf <thomas@cilium.io>
```

**Comment by tgraf** [comment]
This should be resolved in #1085

**Comment by ianvernon** [comment]
Closed, master builds have been stable since #1085 was merged.
